src/Offer/Offer.php

```diff
@@ -722,6 +725,20 @@ protected function importWorkflowStatus(CultureFeed_Cdb_Item_Base $cdbItem): voi
         $this->workflowStatus = $workflowStatus;
     }
 
+    public function updateAvailableFrom(DateTimeInterface $availableFrom): void
+    {
+        if ($availableFrom < new DateTimeImmutable()) {
+            $availableFrom = new DateTimeImmutable();
+        }
+
+        // It is required to use `==` instead of `===` to compare DateTime objects in PHP
+        if ($this->availableFrom !== null && $this->availableFrom == $availableFrom) {
```

Comment:
Why do we still need the null check if we're already comparing with a `DateTimeInterface` instance?

Reply:
```suggestion
        if ($this->availableFrom == $availableFrom) {
```

Reply:
You are right! Although the internal property `availableFrom` can start as `null`, the incoming parameter `availableFrom` can never be `null` because of the `DateTimeInterface` type hint.